adam/learner/attributes.py

```diff
@@ -11,16 +13,89 @@
     PerceptionGraphFromObjectRecognizer,
 )
 from adam.learner.perception_graph_template import PerceptionGraphTemplate
-from adam.learner.subset import AbstractTemplateSubsetLearner
-from adam.learner.surface_templates import STANDARD_SLOT_VARIABLES, SurfaceTemplate
-from adam.learner.template_learner import AbstractTemplateLearner
+from adam.learner.subset import (
+    AbstractTemplateSubsetLearner,
+    AbstractTemplateSubsetLearnerNew,
+)
+from adam.learner.surface_templates import (
+    BoundSurfaceTemplate,
+    SLOT1,
+    STANDARD_SLOT_VARIABLES,
+    SurfaceTemplate,
+)
+from adam.learner.template_learner import (
+    AbstractTemplateLearner,
+    AbstractTemplateLearnerNew,
+)
 from adam.perception import PerceptualRepresentation
 from adam.perception.developmental_primitive_perception import (
     DevelopmentalPrimitivePerceptionFrame,
 )
-from adam.perception.perception_graph import LanguageAlignedPerception, PerceptionGraph
+from adam.perception.perception_graph import PerceptionGraph
+from adam.learner.alignments import (
+    LanguageConceptAlignment,
+    LanguagePerceptionSemanticAlignment,
+    PerceptionSemanticAlignment,
+)
 from attr import attrib, attrs
 from immutablecollections import immutabledict, immutableset
+from vistautils.span import Span
+
+
+@attrs
+class AbstractAttributeTemplateLearnerNew(AbstractTemplateLearnerNew, ABC):
+    def _candidate_templates(
+        self, language_perception_semantic_alignment: LanguagePerceptionSemanticAlignment
+    ) -> AbstractSet[BoundSurfaceTemplate]:
+        ret = []
+        language_concept_alignment = (
+            language_perception_semantic_alignment.language_concept_alignment
+        )
+        # Find all objects we have recognized...
+        for (
+            object_node,
+            span_for_object,
+        ) in language_concept_alignment.node_to_language_span.items():
+            if isinstance(object_node, ObjectSemanticNode):
+                # Any words immediately before them or after them are candidate attributes.
+                # TODO: make issue to track this somewhat English-specific assumption
```

Comment:
Reminder to turn this into an issue. I'm going to mark future ones of these with a 🗒️ emote

Reply:
@spigo900 : Please make the requested issue and add a pointer to it in the code

Reply:
@gabbard Created the issue: #791.